### PR TITLE
WIP-[GH11] CRUD Part 2: POST on Product Detail Page

### PR DESCRIPTION
**Pull request by 1000miles**
This PR does following:

### Fixes:

- corrects `location` props mapping e.g. to `this.state.location.latitude` based on @ninetteadhikari's [feedback](https://github.com/IchiBizz/IchiBizz/pull/20/files#r331744761) 

### Changes:

- converts `condition` from datatype string to `enum` of strings => allows to use radio button and to chose between `used` and `new` on the product detail form, easier to use search filter on it later.
- moves components `ProductsList`, `ProductListStyles`, `AddProduct` to a new `products` folder => `/components/products/*` to have a cleaner structure and adopts import paths of effected files
- partial refactoring and restructuring, e.g. changes initial states and handleSubmits for multiple props

### Introductions:

- sets timestamps for `Product` model => pepares `this.state.created_at` props to render in the front-end later

Fixes Part of GH issue: #11 

---

### Open:

- Left out this [part](https://github.com/IchiBizz/IchiBizz/pull/20/files#r331744783) based on @ninetteadhikari's feedback to check later.

---

@ninetteadhikari @noliko-zwa 

### Please do not merge this PR for now.

This PR will most likely cause merge conflicts since it changes the product model. Let's fix the merge conflicts together tomorrow. :)

### CODE REVIEW

File: client/src/components/products/AddProduct.js [hunk 1]
```diff
@@ -69,7 +74,7 @@ export default class AddProduct extends Component {
           category: [],
```
Comment (ninetteadhikari) [inline client/src/components/products/AddProduct.js]: this needs to be changed to empty string : ""
as it is in the model

File: client/src/components/products/AddProduct.js [hunk 2]
```diff
@@ -78,7 +83,8 @@ export default class AddProduct extends Component {
           },
           availability: null,
           warrantyUntil: null,
-          condition: ""
+          condition: "",
+          createdAt: new Date()
```
Comment (ninetteadhikari) [inline client/src/components/products/AddProduct.js]: this needs to be changed to match with the initial state as 'null'

File: client/src/components/products/ProductsList.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React, { Component } from "react";
 import axios from "axios";
 import { Link } from "react-router-dom";
-import SearchFilter from "./SearchFilter";
+import SearchFilter from "../SearchFilter";
```
Comment (ninetteadhikari) [inline client/src/components/products/ProductsList.js]: this file to be moved to the 'product' folder

**Review by ninetteadhikari — COMMENTED** [review]
Have added some comments. please review

**Review by ninetteadhikari — CHANGES_REQUESTED** [review]
practicing request changes

File: client/src/components/products/AddProduct.js (hunk 1 above)
Comment (1000miles) [reply client/src/components/products/AddProduct.js]: True that. Thanks for the review, I updated the PR. 🎈

File: client/src/components/products/AddProduct.js (hunk 2 above)
Comment (1000miles) [reply client/src/components/products/AddProduct.js]: True that. Thanks for the review, I updated the PR. 🎈

File: client/src/components/products/ProductsList.js (hunk 3 above)
Comment (1000miles) [reply client/src/components/products/ProductsList.js]: Got it, thanks. I updated.

File: client/src/components/products/AddProduct.js (hunk 1 above)
Comment (ninetteadhikari) [reply client/src/components/products/AddProduct.js]: super!!

**Pull request merged**